src/main/kotlin/Calculator.kt

```diff
@@ -0,0 +1,58 @@
+import java.util.LinkedList
+import java.util.Queue
+
+object Calculator {
+
+    fun calEquation(equation: String?): Double {
+        if (equation == null || equation.isEmpty()) {
+            throw IllegalArgumentException("입력 값이 유효하지 않습니다.")
+        }
+
+        val (numList: Queue<Double>, operatorList) = LinkedList<Double>() to mutableListOf<Char>()
+        separateNumAndOpList(equation, numList, operatorList)
+
+        return calOperator(numList, operatorList)
+    }
+
+    fun calOperator(numList: LinkedList<Double>, operatorList: List<Char>): Double {
```

Comment:
이왕 만든 `Operator`를 사용하면 어떨까요?

```suggestion
    fun calOperator(numList: LinkedList<Double>, operatorList: List<Operator>): Double {
```

Reply:
적용하였습니다.
[7e4cae610731399a891c0f1bdca233d0e17406fa]